varats/varats/data/metrics.py

```diff
@@ -132,3 +132,122 @@ def min_max_normalize(values: pd.Series) -> pd.Series:
     max_value = values.max()
     min_value = values.min()
     return tp.cast(pd.Series, (values - min_value) / (max_value - min_value))
+
+
+class ConfusionMatrix:
+    """
+    Helper class to automatically calculate classification results.
+
+                        |  Predicted Positive (PP)  |  Predicted Negative (PN)
+    --------------------|---------------------------|--------------------------
+    Actual Positive (P) |  True Positive      (TP)  |  False Negative     (FN)
+    Actual Negative (N) |  False Positive     (FP)  |  True Negative      (TN)
+
+    Reference: https://en.wikipedia.org/wiki/Precision_and_recall
+    """
+
+    def __init__(
+        self, actual_positive_values: tp.List[tp.Any],
+        actual_negative_values: tp.List[tp.Any],
+        predicted_positive_values: tp.List[tp.Any],
+        predicted_negative_values: tp.List[tp.Any]
+    ) -> None:
+        self.__actual_positive_values = actual_positive_values
+        self.__actual_negative_values = actual_negative_values
+        self.__predicted_positive_values = predicted_positive_values
+        self.__predicted_negative_values = predicted_negative_values
+
+    ###################
+    # Base values
+
+    @property
+    def P(self) -> int:  # pylint: disable=C0103
+        return len(self.__actual_positive_values)
+
+    @property
+    def N(self) -> int:  # pylint: disable=C0103
+        return len(self.__actual_negative_values)
+
+    @property
+    def PP(self) -> int:  # pylint: disable=C0103
+        return len(self.__predicted_positive_values)
+
+    @property
+    def PN(self) -> int:  # pylint: disable=C0103
+        return len(self.__predicted_negative_values)
+
+    ###################
+    # Combined values
+
+    @property
+    def TP(self) -> int:  # pylint: disable=C0103
+        return len(
+            set(self.__actual_positive_values
+               ).intersection(self.__predicted_positive_values)
+        )
```

Comment:
```suggestion
    def TP(self) -> tp.Set[tp.Any]:  # pylint: disable=C0103
        return set(self.__actual_positive_values
               ).intersection(self.__predicted_positive_values)
```
We should add a way to retrieve the classified elements instead of just their number. Either we could change the interface as suggested or introduce seperate methods.

Reply:
Why would you need the elements? I did not plan to allow users to access them, as (1) the values should be enough for a eval and (2) during construction, the user already passes correctly attributed lists to us.
What you be your use case where you need access to the elements?

Reply:
I would like to inspect/print the FP and FN elements. If we don't expose them, I have to build the intersections myself. That duplicates code and kind of defeats the purpose of using this confusion matrix implementation. I.e., if I have to classify the elements myself anyway, why should I put them again in the attributed lists just to be able to instantiate a `ConfusionMatrix`? If I have my TP/TN/FP/FN lists already, I can use these directly to compute the needed interpretations.

Reply:
Yes, but you don't have to implement all the different values. If you need the TP values a simple intersection is enough :thinking: